arxiv/base/templates/base/header.html

```diff
@@ -13,6 +13,24 @@ <h1 class="level-item"><a href="{{ url_for('home') }}" aria-label="arxiv-logo">a
   {{ macros.compactsearch('level-right') }}
 </div> <!-- closes identity -->
 
-<!--  # TODO: reintroduce this once we have access to the user's session.
-  <div class="user-tools box is-pulled-right" role="navigation" aria-label="User menu"><a href="{{ url_for('login') }}">Login</a> | <a href="{{ url_for('account') }}">My Account</a> | <a href="{{ url_for('logout') }}">Logout</a></div>
--->
+<div class="is-pulled-right">
+  <div class="user-tools box level-item" role="navigation" aria-label="User menu">
+    {% if request.session and request.session.user %}Logged in as <strong>{{ request.session.user.username }}</strong> | <a href="{{ url_for('account') }}">My Account</a> | <a href="{{ url_for('logout') }}">Logout</a>{% else %}<a href="{{ url_for('login') }}">Login</a>{% endif %}
+  </div>
+</div>
+
+<div class="container is-fluid notifications">
+    {% with messages = get_flashed_messages(with_categories=true) %}
+      {% if messages %}
+        {% for category, message in messages %}
+          <div class="notification is-{{ category }}">
+            {% if message.dismissable %}<button class="delete notification-dismiss"></button>{% endif %}
+            {% if message.title %}<div class="notification-title">{{ message.title }}</div>{% endif %}
```

Comment:
Yeah, this should probably be at some heading level for a11y. I'll review onscreen to figure out which one it should be to avoid skipping heading levels.